astropy/tests/helper.py

```diff
@@ -414,7 +414,13 @@ def generic_recursive_equality_test(a, b, class_history):
     for key in dict_a:
         assert key in dict_b,\
           f"Did not pickle {key}"
-        if hasattr(dict_a[key], '__eq__'):
```

Comment:
I *think* the intent here was to do what I now added. But what was here would not have worked. At least, I think everything in python has an `__eq__` method!